newIDE/app/src/AssetStore/AssetStoreSearchFilter.js

```diff
@@ -261,6 +261,155 @@ export class ColorAssetStoreSearchFilter
   }
 }
 
+// TODO
+const toAssetStoreType = (type: string) => {
+  return type === 'Sprite' ? 'sprite' : type;
+};
+
+// Thematic tags are noise for asset swapping as changing the theme may be what
+// users are looking for.
+const ignoredThematicTags = new Set([
+  'top-down',
+  'isometric',
+  'side view',
+  'pixel art',
+  'pirate',
+  'space',
+  'sea',
+  'city',
+  'medieval',
+  'nature',
+  'forest',
+]);
+
+export class AssetSwappingAssetStoreSearchFilter
+  implements SearchFilter<AssetShortHeader> {
+  isEnabled: boolean;
+  objectType: string;
+  hasSeveralState: boolean;
+  other: AssetShortHeader | null;
+  tags: Array<string>;
+
+  constructor(
+    object: gdObject | null = null,
+    assetShortHeader: AssetShortHeader | null = null
+  ) {
+    if (object) {
+      this.isEnabled = true;
+      this.objectType = assetShortHeader
+        ? assetShortHeader.objectType
+        : toAssetStoreType(object.getType());
+      this.hasSeveralState = object.getConfiguration().getAnimationsCount() > 0;
+      this.other = assetShortHeader;
+      // The asset pack tag is not relevent.
+      this.tags = assetShortHeader
+        ? assetShortHeader.tags
+            .slice(1)
+            .filter(tags => !ignoredThematicTags.has(tags))
+        : [];
+    } else {
+      this.isEnabled = false;
+      this.objectType = '';
+      this.hasSeveralState = false;
+      this.other = null;
+      this.tags = [];
+    }
+  }
+
+  getPertinence(searchItem: AssetShortHeader): number {
```

Comment:
This would be worth a unit test to check something like:
```
const assetSwappingAssetStoreSearchFilter = new AssetSwappingAssetStoreSearchFilter(someObject, fakeAssetShortHeader);
expect(assetSwappingAssetStoreSearchFilter.getPertinence(otherFakeAssetShortHeader)).toBe(...)
```

Reply:
I'm not sure it's worth it because there is no exact meaning behind the pertinence value. The tests would go to waste every time the evaluation is refined.
Even the sort is fuzzy for efficiency reason.

Reply:
agree 👍